addon/initializers/embedded.ts

```diff
@@ -96,6 +98,8 @@ function normalizeConfig(userConfig: GivenConfig): ObjectConfig {
 
 export function initialize(application: Application): void {
   const env = application.resolveRegistration('config:environment')
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore: https://github.com/typed-ember/ember-cli-typescript/issues/1471
   const embeddedConfig: ObjectConfig = normalizeConfig(env.embedded)
```

Comment:
I think here we could directly type-assert what we get from `config:environment`, it would not be a big deal:

```suggestion
  const env = application.resolveRegistration('config:environment') as { embedded?: GivenConfig }
  const embeddedConfig: ObjectConfig = normalizeConfig(env.embedded)
```

And the TS error on `env.embedded` would go away.

I believe there is nothing wrong with casting (`as ...`) since we already assume in our code that users that consume our add-on could pass (or not pass) a configuration object potentially nullish / boolean / correct (since type `GivenConfig`).

Reply:
Way better like this, thanks!